pandas/core/dtypes/dtypes.py

```diff
@@ -2326,3 +2347,29 @@ def __from_arrow__(self, array: pa.Array | pa.ChunkedArray):
         array_class = self.construct_array_type()
         arr = array.cast(self.pyarrow_dtype, safe=True)
         return array_class(arr)
+
+    def _find_compatible_dtype(self, item: Any) -> tuple[DtypeObj, Any]:
+        if isinstance(item, pa.Scalar):
```

Comment:
what happens if item is null? The pyarrow null

Reply:
IIUC pyarrow nulls are now typed.  id prefer to be strict about making these match, but dont care _that_ much.  am hoping @jorisvandenbossche will weigh in